test/scenarios/applicationmonitoring/dataingest.go

```diff
@@ -104,26 +104,27 @@ func deploymentPodsHaveOnlyDataIngestInitContainer(sampleApp sampleapps.SampleAp
 
 		require.NoError(t, err)
 
-		err = query.ForEachPod(assessDeploymentHasDataIngestFile(t, environmentConfig.Client().RESTConfig(), sampleApp.Name()))
+		err = query.ForEachPod(assessDeploymentHasDataIngestFile(ctx, t, environmentConfig.Client().Resources(), sampleApp.Name()))
 
 		require.NoError(t, err)
 
 		return ctx
 	}
 }
 
-func assessDeploymentHasDataIngestFile(t *testing.T, restConfig *rest.Config, deploymentName string) deployment.PodConsumer {
+func assessDeploymentHasDataIngestFile(ctx context.Context, t *testing.T, resource *resources.Resources, deploymentName string) deployment.PodConsumer {
 	return func(pod corev1.Pod) {
-		dataIngestMetadata := getDataIngestMetadataFromPod(t, restConfig, pod)
+		dataIngestMetadata := getDataIngestMetadataFromPod(ctx, t, resource, pod)
 
 		assert.Equal(t, dataIngestMetadata.WorkloadKind, "Deployment")
 		assert.Equal(t, dataIngestMetadata.WorkloadName, deploymentName)
 	}
 }
 
-func getDataIngestMetadataFromPod(t *testing.T, restConfig *rest.Config, dataIngestPod corev1.Pod) metadata {
-	query := pod.NewExecutionQuery(dataIngestPod, dataIngestPod.Spec.Containers[0].Name, shell.ReadFile(metadataFile)...)
-	result, err := query.Execute(restConfig)
+func getDataIngestMetadataFromPod(ctx context.Context, t *testing.T, resource *resources.Resources, dataIngestPod corev1.Pod) metadata {
+	readMetadataCommand := shell.ReadFile(metadataFile)
+	executeQuery := pod.NewExecutionQuery(ctx, resource, dataIngestPod, dataIngestPod.Spec.Containers[0].Name, readMetadataCommand...)
+	result, err := executeQuery.Execute()
```

Comment:
Same, a  convenience function could be nice. The query itself is never used elsewhere in this scope.